### PR TITLE
Update to include new Enum functionality + min/max function updates for Elixir 1.4

### PR DESCRIPTION
**Pull request by ybur-yug**
Though 1.4 isn't officially released I figured it would be good to clarify about `min/1` vs `min/2` and `max/1` vs `max/2`, as well as update for the deprecation of `uniq/1` and include the new `map_every/3` function that is included in the release, as it meshes well with the `chunk` and `chunk_by` pieces quite well.

We should probably wait to merge this until the official release as to not confuse anyone with features not yet available without installing from source. I added the `fix` label since `uniq` is being deprecated.

### CODE REVIEW

File: lessons/basics/enum.md [hunk 1]
```diff
@@ -83,22 +92,36 @@ iex> Enum.map([0, 1, 2, 3], fn(x) -> x - 1 end)
 
 ### min
 
-Find the `min` value in our collection:
+`min/1` finds the `min` value in the collection.
```
Comment (brain-geek) [inline lessons/basics/enum.md]: What's the point of changing ':' to '.' here?

File: lessons/basics/enum.md [hunk 2]
```diff
@@ -83,22 +92,36 @@ iex> Enum.map([0, 1, 2, 3], fn(x) -> x - 1 end)
...
 
-Returns the `max` value in the collection:
+`max/1` returns the `max` value in the collection:
 
 ```elixir
 iex> Enum.max([5, 3, 0, -1])
 5
 ```
 
+`max/2` does the same, and behaves as `min/2` does as well, allowing us to pass an anonymous function to provide a default value to `Enum`.
```
Comment (brain-geek) [inline lessons/basics/enum.md]: Same here with trailing dot - usually we use ':'.

File: lessons/basics/enum.md (hunk 1 above)
Comment (ybur-yug) [inline lessons/basics/enum.md]: No point, just me being silly & missing a small detail. 👍 will change in both locations & push back up sometime today after I finish up the days work.

**Comment by ybur-yug** [comment]
@brain-geek, updated. Thanks!

File: lessons/basics/enum.md [hunk 3]
```diff
@@ -59,6 +59,15 @@ iex> Enum.chunk_by(["one", "two", "three", "four", "five", "six"], fn(x) -> Stri
 [["one", "two"], ["three"], ["four", "five"], ["six"]]
 ```
 
+### map_every
+
+Sometimes chunking out a collection isn't enough for exactly what we may need. If this is the case, `map_every/3` can be very useful to hit only specific items if your collection has an ordering that such is necessary or useful.
```
Comment (brain-geek) [inline lessons/basics/enum.md]: I missed this `:` <-> `.` previous time. :)

**Comment by brain-geek** [comment]
@ybur-yug one more place. :)

**Comment by ybur-yug** [comment]
@brain-geek got it!

File: lessons/basics/enum.md (hunk 1 above)
Comment (brain-geek) [inline lessons/basics/enum.md]: And what about this one? :)

**Comment by brain-geek** [comment]
@ybur-yug last one! Otherwise, everything is OK to merge.

File: lessons/basics/enum.md (hunk 1 above)
Comment (ybur-yug) [inline lessons/basics/enum.md]: Gah! fixing.

**Comment by ybur-yug** [comment]
Ok if you wanna give it one final lookover I think thats all of them but I also have not excelled at noticing small details today it seems @brain-geek

**Comment by brain-geek** [comment]
@ybur-yug Feel free to merge.

**Comment by ybur-yug** [comment]
@brain-geek I'll make an issue for translation, and once 1.4 is out I'll officially merge since these functionalities/changes are not present unless running it which requires a build from source on master right now.

Especially since this is in the basics section, I don't want to confuse anyone who comes in on a blank slate.

**Comment by brain-geek** [comment]
I've pushed my commit to this branch.

Just one more thing before merge: @ybur-yug , please squash first few commits into one (some of them don't have any semantic value) .

**Comment by brain-geek** [comment]
@ybur-yug one of the commits in the branch still has ` lol cmon bobby ` title. :)

**Comment by ybur-yug** [comment]
@brain-geek I'll squash my stuff and keep the others here in a bit today. Sorry I just got to this, been busy the last week

**Comment by ybur-yug** [comment]
Going to clean the history here then merge now that 1.4 is out

**Comment by doomspork** [comment]
Awesome!  If we can get this out plus the staged new lessons, we could do a big ol release announcement to kick off 2017.

**Comment by brain-geek** [comment]
Elixir 1.4 has been released finally. Let's merge this after fixing history

**Comment by ybur-yug** [comment]
@brain-geek I pushed up after squashing my crap commits, if it looks good to you feel free to merge.

**Comment by brain-geek** [comment]
Awesome. Thanks, @ybur-yug !

We can finally merge and close this one. :)

**Pull request merged**